### ISSUE TITLE
Ghost field added to FHIR mockdoc, which gets duplicated `@` XML attribute prefix as it reloads

### ISSUE THREAD

**Issue by igarashitm** [atlasmap/atlasmap#1518]
![reload-attr-dup](https://user-images.githubusercontent.com/265462/69424195-04308c00-0cf6-11ea-80cf-f7c049d74c4a.gif)

**Comment by igarashitm** [comment]
It's entering here
https://github.com/atlasmap/atlasmap/blob/4f8901fd59b4a374c829168a89304490ffa45860/ui/src/app/lib/atlasmap-data-mapper/utils/mapping-util.ts#L94

which means this doesn't return proper field
https://github.com/atlasmap/atlasmap/blob/4f8901fd59b4a374c829168a89304490ffa45860/ui/src/app/lib/atlasmap-data-mapper/utils/mapping-util.ts#L80

So my guess is that the mapping deserialization is happening before document definition is completely built.

**Comment by igarashitm** [comment]
TLDR I put this on hold until React migration is settled.

This `fetchDocuments()` is async work
https://github.com/atlasmap/atlasmap/blob/c9977f167d3926be913457e342f3ea896e084173/ui/src/app/lib/atlasmap-data-mapper/services/initialization.service.ts#L203

And it's possible that this `fetchMappings()` starts to deserialize mappings before document initialization by `fetchDocuments()` is completed.

I thought we can subscribe `initializationStatusChanged$` and wait for `ConfigModel.initCfg.initialized` comes to true, but then it doesn't happen until field action service is initialized.
https://github.com/atlasmap/atlasmap/blob/c9977f167d3926be913457e342f3ea896e084173/ui/src/app/lib/atlasmap-data-mapper/services/initialization.service.ts#L534

The field action service initialization is delayed on purpose to wait for custom field actions to be loaded when .adm is uploaded. We have to untangle this initialization statuses, basically in following order:
1. library upload and classpath refresh should happen first, then
1. document initialization
1. field action service
1. mappings

**Comment by igarashitm** [comment]
A good news is that, this is standalone only issue, as in Syndesis mapping definition is passed in in UI, but not loading from .adm file.

**Comment by igarashitm** [comment]
This issue is not solved yet. Note that the ghost field `/@value` is added to the FHIR document. Also the real attribute doesn't get `@` prefix it seems.
![output](https://user-images.githubusercontent.com/265462/102078545-6f438800-3e4e-11eb-8eff-90b679ada014.gif)

While we need a way to distinguish element and attribute for XML document, I don't think adding `@` prefix to the field name is the best way. Instead for example, we can add an icon that indicates it's an attribute.

`/myelement/@mychild` and `/myelement/mychild` have to show differently in UI.
```xml
<myelement mychild="one">
  <mychild>the other</mychild>
</myelement>
```

**Comment by pleacu** [comment]
So this is the correct mapping:

<pre>      &quot;mapping&quot; : [ {
        &quot;jsonType&quot; : &quot;io.atlasmap.v2.Mapping&quot;,
        &quot;inputField&quot; : [ {
          &quot;jsonType&quot; : &quot;io.atlasmap.xml.v2.XmlField&quot;,
          &quot;docId&quot; : &quot;fhir-patient-246bb727-92e7-43e1-8da5-50abf8be1d17&quot;,
          &quot;path&quot; : &quot;/tns:Patient/tns:id/@value&quot;,
          &quot;fieldType&quot; : &quot;STRING&quot;,
          &quot;name&quot; : &quot;value&quot;,
          &quot;attribute&quot; : false
        } ],
        &quot;outputField&quot; : [ {
          &quot;jsonType&quot; : &quot;io.atlasmap.xml.v2.XmlField&quot;,
          &quot;docId&quot; : &quot;fhir-patient-72a005cc-6a45-4573-89f7-bad2ecc9dd59&quot;,
          &quot;path&quot; : &quot;/tns:Patient/tns:id/@value&quot;,
          &quot;fieldType&quot; : &quot;STRING&quot;,
          &quot;name&quot; : &quot;value&quot;,
          &quot;attribute&quot; : false
        } ],
        &quot;id&quot; : &quot;mapping.654913&quot;
      } ]
</pre>
So specifically we want to continue to decorate the path with the `@value` - i.e. `"path" : "/tns:Patient/tns:id/@value"` but we do *not* want to display the `@value`  just `tns:value` - is that correct?

**Comment by igarashitm** [comment]
Let's see one by one
1. Now it might be better to file separate issues for the `@` prefix. The most problematic issue here is the ghost field, non-existing field is added to the Document as it reloads, and eventually breaks the mapping.
1. The mapping definition in https://github.com/atlasmap/atlasmap/issues/1518#issuecomment-744631528 looks correct
1. `Field.attribute` should be true in theory. It might not be used yet though. Inspection already sets it true https://github.com/atlasmap/atlasmap/issues/1732
1. The path has to have `@` prefix for the attribute
1. It was supposed to be shown as `@tns:value` or `@value` in the document tree, but now it is not. I think it's a bug since IIRC it used to be like that, and we need something to distinguish the attribute from the element with the same name (see https://github.com/atlasmap/atlasmap/issues/1518#issuecomment-744394433)
1. However, adding `@` prefix directly to the display name doesn't look the best to indicate the attribute. We can improve it by showing an icon instead.
1. Even that, `Field.name` should not have `@` prefix. This `@` prefix issues are only about how to show in UI.
